include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -979,6 +1033,268 @@ namespace internal
+      {
+        Assert(false, ExcNotImplemented());
+      }
+
+      /**
+       * Given a FEValuesBase object, extract the gradients on the present
+       * cell from the vector we actually store. This function does the same
+       * as the one above but for vector-valued finite elements.
+       */
+      virtual void
+      get_function_gradients(
+        const FEValuesBase<DoFHandlerType::dimension,
+                           DoFHandlerType::space_dimension>
+          & /*fe_patch_values*/,
+        const ComponentExtractor /*extract_component*/,
+        std::vector<std::vector<Tensor<1, DoFHandlerType::space_dimension>>>
+          & /*patch_gradients_system*/) const override
+      {
+        Assert(false, ExcNotImplemented());
+      }
+
+
+      /**
+       * Given a FEValuesBase object, extract the second derivatives on the
+       * present cell from the vector we actually store.
+       */
+      virtual void
+      get_function_hessians(
+        const FEValuesBase<DoFHandlerType::dimension,
+                           DoFHandlerType::space_dimension>
+          & /*fe_patch_values*/,
+        const ComponentExtractor /*extract_component*/,
+        std::vector<Tensor<2, DoFHandlerType::space_dimension>>
+          & /*patch_hessians*/) const override
+      {
+        Assert(false, ExcNotImplemented());
+      }
+
+      /**
+       * Given a FEValuesBase object, extract the second derivatives on the
+       * present cell from the vector we actually store. This function does
+       * the same as the one above but for vector-valued finite elements.
+       */
+      virtual void
+      get_function_hessians(
+        const FEValuesBase<DoFHandlerType::dimension,
+                           DoFHandlerType::space_dimension>
+          & /*fe_patch_values*/,
+        const ComponentExtractor /*extract_component*/,
+        std::vector<std::vector<Tensor<2, DoFHandlerType::space_dimension>>>
+          & /*patch_hessians_system*/) const override
+      {
+        Assert(false, ExcNotImplemented());
+      }
+
+      /**
+       * Return whether the data represented by (a derived class of) this object
+       * represents a complex-valued (as opposed to real-valued) information.
+       */
+      virtual bool
+      is_complex_valued() const override
+      {
+        return false;
+      }
+
+      /**
+       * Clear all references to the vectors.
+       */
+      virtual void
+      clear() override
+      {
+        vectors = nullptr;
+      }
+
+      /**
+       * Determine an estimate for the memory consumption (in bytes) of this
+       * object.
+       */
+      virtual std::size_t
+      memory_consumption() const override
+      {
+        return sizeof(vectors);
+      }
+
+    private:
+      const MGLevelObject<VectorType> *vectors;
+    };
+
+
+
+    template <typename DoFHandlerType, typename VectorType>
+    void
+    MGDataEntry<DoFHandlerType, VectorType>::get_function_values(
```

Comment:
Are these functions essentially copies of the same functions for non-mg vectors with the exception that
* we call `cell->get_mg_dof_indices()`
* what else?
Maybe say something to this effect at the top of the function just to make it clear?

Reply:
Not quite. For the active levels, we call FEValues::get_function_values() which I can not use.